modules/madvertiseBidAdapter.js

```diff
@@ -0,0 +1,131 @@
+
+export const spec = {
+  code: 'madvertise',
+  aliases: ['madvertise'],
+  /**
+   * @param {object} bid
+   * @return boolean
+   */
+  isBidRequestValid: function (bid) {
+    if (typeof bid.params !== 'object') {
+      return false;
+    }
+    if (!Array.isArray(bid.sizes) || bid.sizes.length == 0) {
+      return false;
+    } else {
+      var sizes = [];
+      for (var i = 0; i < bid.sizes.length; i++) {
+        if (Array.isArray(bid.sizes[i]) && bid.sizes[i].length == 2) {
+          sizes.push(bid.sizes[i]);
+        }
+      }
+      if (sizes.length == 0) {
+        return false;
+      }
+    }
+
+    if (typeof bid.params.floor == 'undefined' || parseFloat(bid.params.floor) < 0.01) {
+      bid.params.floor = 0.01;
+    }
+
+    return typeof bid.params.s != 'undefined';
+  },
+  /**
+   * @param {BidRequest[]} bidRequests
+   * @return ServerRequest[]
+   */
+  buildRequests: function (bidRequests) {
+    return bidRequests.map(bidRequest => {
+      bidRequest.startTime = new Date().getTime();
+
+      // non-video request builder
+      var src = '?rt=bid_request&v=1.0';
+
+      for (var i = 0; i < bidRequest.sizes.length; i++) {
+        if (Array.isArray(bidRequest.sizes[i]) && bidRequest.sizes[i].length == 2) {
+          src = src + '&sizes[' + i + ']=' + bidRequest.sizes[i][0] + 'x' + bidRequest.sizes[i][1];
+        }
+      }
+
+      utils._each(bidRequest.params, (item, key) => src = src + '&' + key + '=' + item);
+
+      if (typeof bidRequest.params.u == 'undefined') {
+        src = src + '&u=' + navigator.userAgent;
+      }
+
+      src = src + _getSeenAd();
+
+      return {
+        method: 'GET',
+        url: MADVERTISE_ENDPOINT + src,
+        options: {withCredentials: false},
+        bidId: bidRequest.bidId
+      };
+    });
+  },
+  /**
+   * @param {*} responseObj
+   * @param {BidRequest} bidRequest
+   * @return {Bid[]} An array of bids which
+   */
+  interpretResponse: function (responseObj, bidRequest) {
+    responseObj = responseObj.body;
+    // check overall response
+    if (responseObj == null || typeof responseObj !== 'object' || !responseObj.hasOwnProperty('ad')) {
+      return [];
+    }
+
+    _setSeenAd(responseObj.creativeId);
+
+    let bid = {
+      requestId: bidRequest.bidId,
+      cpm: responseObj.cpm,
+      width: responseObj.Width,
+      height: responseObj.height,
+      ad: responseObj.ad,
+      ttl: responseObj.ttl,
+      creativeId: responseObj.creativeId,
+      netRevenue: responseObj.netRevenue,
+      currency: responseObj.currency,
+      dealId: responseObj.dealId
+    };
+    return [bid];
+  },
+  getUserSyncs: function (syncOptions) {
+  }
+};
+
+function _getSeenAd() {
+  var cookies = document.cookie.split(';');
+  var src = '';
+  for (var i = 0; i < cookies.length; i++) {
+    var cookie = cookies[i].trim();
+    if (cookie.match(/seenad\[\d+\]=\d+/)) {
+      src += '&' + cookie;
+    }
+  }
+  return src;
+}
+
+function _setSeenAd(adid) {
```

Comment:
why are you saving this data in the publisher domain?

Reply:
For capping, we send adId display on device. We need to remove this feature ?

Reply:
it's ok or not for _setSeenAd ?

Reply:
Please remove it for now

Reply:
done.